packages/@react-spectrum/table/src/Table.tsx

```diff
@@ -379,15 +403,17 @@ function TableColumnHeader({column}) {
               'react-spectrum-Table-cell',
               {
                 'react-spectrum-Table-cell--alignCenter': columnProps.align === 'center' || column.colspan > 1,
-                'react-spectrum-Table-cell--alignEnd': columnProps.align === 'end'
+                'react-spectrum-Table-cell--alignEnd': columnProps.align === 'end',
+                'react-spectrum-Table-cell--hideHeader': columnProps.hideHeader
               }
             )
           )
         }>
-        {column.rendered}
+        {!columnProps?.hideHeader && column.rendered}
```

Comment:
weird question, but in what case are columnProps undefined?

Reply:
This is just a fail safe, check. I will remove it if its not something that should be added